### PR TITLE
Update compaction, remove accounts.v3

### PR DESCRIPTION
**Pull request by abramin**

### CODE REVIEW

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: prod-aws
Module: account-platform/kafka-shared
Path: prod-aws/account-platform/
Commit ID: 920a3084c9b8975a49fd305f81d9303589014098
Requested At: 2024-08-30T10:16:58Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan prod-aws/account-platform/` as comment.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: prod-aws
Module: account-platform/kafka-shared
Path: prod-aws/account-platform/
Commit ID: 1e8f6c2ba2b2ec6e5872ba80d8f82a48230a9d7a
Requested At: 2024-08-30T10:19:15Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan prod-aws/account-platform/` as comment.

File: prod-aws/account-platform/topics.tf [hunk 1]
```diff
@@ -24,3 +23,116 @@ resource "kafka_topic" "account_exceptions_events" {
...
+  to = kafka_topic.legacy_account_change_events_compacted
+  id = "legacy.account.change.events.compacted"
+}
+resource "kafka_topic" "legacy_account_change_events_compacted" {
+  config = {
+    "cleanup.policy"        = "compact"
+    "max.compaction.lag.ms" = "604800000"
+  }
+  name               = "legacy.account.change.events.compacted"
+  partitions         = 50
```
Comment (Jarota) [inline prod-aws/account-platform/topics.tf:80]: Not necessarily related to the PR but 50 partitions sounds like a lot, just curious how you guys decided on this config? 🙇

**Review by Jarota — APPROVED** [review]

**Review by matiit — APPROVED** [review]

File: prod-aws/account-platform/topics.tf (hunk 1 above)
Comment (abramin) [reply prod-aws/account-platform/topics.tf:80]: It´s a good question to keep asking. 
We don´t necessarily need 50 for this one.
But for very large topics with a lot of consumers like legacy.account.events, it helps speed up consumption.

I think (maybe) when compacting with a relay you need to match partitions in source and target, but can´t remember if that´s true.

Before we migrate these topics let´s revisit partition numbers

**Pull request merged**